ivy/functional/ivy/experimental/creation.py

```diff
@@ -475,3 +476,97 @@ def tril_indices(
 
     """
     return current_backend().tril_indices(n_rows, n_cols, k, device=device)
+
+
+@to_native_arrays_and_back
+@handle_out_argument
+@infer_dtype
+@infer_device
+@handle_nestable
+@handle_exceptions
+@handle_array_like_without_promotion
+def eye_like(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    k: int = 0,
+    dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+    device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """Returns a array filled with ones on the k diagonal and zeros elsewhere. having
+    the same ``shape`` as an input array ``x``.
+
+
+    Parameters
+    ----------
+    x
+         input array from which to derive the output array shape.
+    k
+        index of the diagonal. A positive value refers to an upper diagonal, a negative
+        value to a lower diagonal, and 0 to the main diagonal. Default: ``0``.
+    dtype
+        output array data type. If dtype is None, the output array data type must be the
+        default floating-point data type. Default: ``None``.
+    device
+        the device on which to place the created array.
+    out
+        optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        an array having the same shape as ``x`` and filled with ``ones`` in diagonal ``k``
+        and ``zeros`` elsewhere.
+
+    This function conforms to the `Array API Standard
```

Comment:
This function doesn't conform to Array API. Feel free to refer to onnx docs and update docstrings.
https://github.com/onnx/onnx/blob/main/docs/Operators.md#EyeLike

After updating docstrings the changes will be good to be merged🙂

Reply:
sorry, I didn't notice it. I changed it now, Should I add also onnx docs link or the description is enough?

Reply:
Description is enough